### PR TITLE
Moves and updates basic development workflow

### PR DESCRIPTION
**Pull request by lukpueh**
@awwad has created a nice document about our lab development workflow in https://github.com/secure-systems-lab/code-style-guidelines/pull/8. I think this repo is better fit for such instructions. So after plenty of rebasing and cherry picking I moved his document over here.

While doing that I updated the basic workflow and stripped out some WIP *further reading* sections and added them to a new document available in the [dev-workflow-misc branch](https://github.com/secure-systems-lab/lab-guidelines/tree/dev-workflow-misc), which still needs some work.

I think for now, we should be fine with this basic description.  But, please feel free to review, comment and/or change, ...

### CODE REVIEW

**Comment by awwad** [comment]
I hijacked this a bunch, as discussed. 🙊 

What do you think of the direction taken, @lukpueh ?

**Comment by lukpueh** [comment]
I would say the overall workflow in  `dev-workflow.md` and the commit section in `commits.md` are pretty solid. Let's merge and create a separate PR for an update of the *"Pull Request guidelines"* which I have labeled *(WIP)* for now. What does everyone think?

**Comment by vladimir-v-diaz** [comment]
I think it's fine to merge.

Is there a reason why this pull request must be merged now?

**Comment by lukpueh** [comment]
The reason why I'd like to merge now is that I've repeatedly pointed students and co-workers to this PR (because IMHO it already gives an excellent overview of our typical workflow). 
So I thought merging would acknowledge that these guidelines are already valid. And it would also give us a base for all sorts of enhancements, e.g. [more details on PRs](a8ed96e62493e5f11bdd49d96d67abaab6ddf70a), [linking to GitHub templates](https://github.com/secure-systems-lab/github-templates), [including git and GitHub tutorials](https://github.com/secure-systems-lab/code-style-guidelines/pull/10), ...

File: commits.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+to have a good sense of what's being changed. For this reason and others
+(cherry-picking, organization, etc.), this means that the substance
+of each commit should be something you can summarize in a line. If you've made
+changes to a module to provide support for some new file format, for example,
+you should try to keep those changes in separate commits from completely
+unrelated changes (for example, adding testing for an already-supported
+file format). You may even want to keep changes that different on distinct
+branches and make distinct pull requests for them.
+
+#### Separate Moving and Modifying Files
```
Comment (aaaaalbert) [inline commits.md:20]: I don't think GitHub has the describe shortcomings actually. I happened to rename and edit a file at some point, and the GitHub diff looks just fine, and the diff box header shows the rename: https://github.com/SeattleTestbed/repy_v2/pull/134/files#diff-08d644fdf23e49274b1b1fb57fb15bb1 .

(Also, I'm not sure about unconditionally separating moves and edits for the other reasons given. The interdependency of not-yet reviewed PRs can be a drag.)

**Comment by lukpueh** [comment]
I'd say we have something we can use. Let's merge and fix/add contents as we move forward.

**Comment by vladimir-v-diaz** [comment]
I don't think the README links to the dev workflow.

**Comment by lukpueh** [comment]
I think [it does](https://github.com/secure-systems-lab/lab-guidelines/blame/master/README.md#L16-L17).

**Pull request merged**